tslearn/svm.py

```diff
@@ -40,13 +40,18 @@ def _preprocess_sklearn(self, X, y=None, fit_time=False):
         else:
             X, y = check_X_y(X, y, allow_nd=True,
                              force_all_finite=force_all_finite)
-        X = check_dims(X, X_fit_dims=None)
         X = to_time_series_dataset(X)
 
         if fit_time:
             self._X_fit = X
             self.gamma_ = gamma_soft_dtw(X)
             self.classes_ = numpy.unique(y)
+        else:
```

Comment:
To reduce the LoC, you could set `check_n_features_only` to `self.kernel in VARIABLE_LENGTH_METRICS` to avoid an if/else. Of course, this reduces the readability a bit.

Reply:
Makes sense. Will do.